src/components/Login.tsx

```diff
@@ -35,9 +38,67 @@ const Login = (props: Props) => {
     setPassword(value);
   }
 
+  const onFirstNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const { value } = event.target;
+    setFirstName(value);
+  }
+
+  const onLastNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const { value } = event.target;
+    setLastName(value);
+  }
+
   const generateAgent = async (event: React.SyntheticEvent) => {
     setLoading(true);
     let agentStatus = await ad4mClient.agent.generate(password);
+    const agentPerspective = await ad4mClient.perspective.add(
+      "My perspective"
+    );
+    const links = [];
+
+    if (firstName) {
+      const link = await ad4mClient.perspective.addLink(
+        agentPerspective.uuid,
+        new Link({
+          source: 'ad4m://profile',
+          target: firstName,
+          predicate: 'sioc://has_firstname'
+        })
+      );
+
+      links.push(link);
+    }
+
+    if (lastName) {
+      const link = await ad4mClient.perspective.addLink(
+        agentPerspective.uuid,
+        new Link({
+          source: 'ad4m://profile',
+          target: lastName,
+          predicate: 'sioc://has_lastname'
+        })
+      );
+
+      links.push(link)
+    }
+
+    const cleanedLinks = [];
+
+    for (const link of links) {
+      const newLink = JSON.parse(JSON.stringify(link));
+      newLink.__typename = undefined;
+      newLink.data.__typename = undefined;
+      newLink.proof.__typename = undefined;
```

Comment:
I don't get why is these three lines needed, could you elaborate a bit?
I guess it's maybe required in GraphQL schema, can we move it to backend (ad4m executor) instead?

Reply:
yes, so we need to explicitly set it to undefined or the ad4mClient throw's an error as the type doesn't match